src/napari_matplotlib/histogram.py

```diff
@@ -65,20 +70,25 @@ def draw(self) -> None:
 
         # Important to calculate bins after slicing 3D data, to avoid reading
         # whole cube into memory.
-        bins = np.linspace(np.min(data), np.max(data), 100)
+        if data.dtype.kind in {"i", "u"}:
+            # Make sure integer data types have integer sized bins
+            step = 1 + abs(np.max(data) - np.min(data)) // 100
```

Comment:
setting the step in this way gives bins with a step size of 3 for the 2D histogram test dataset (whereas previously a step size of 2 was used). Perhaps ensuring `step` is at least 1 is a better way to do it:
```suggestion
            step = abs(np.max(data) - np.min(data)) // 100
            step = max(step, 1)
```

Reply:
👍